docs/source/smooth_quant.md

```diff
@@ -0,0 +1,88 @@
+
+## Quantization Fundamentals
+
+Quantization convert the floating point matrix to an integer matrix.  `Affine quantization` and `Scale quantization`, also called `asymmetric quantization` and `symmetric quantization`, are two common range mapping techniques used in tensor conversion between different data types.
+
+**Asymmetric Quantization**
+
+In asymmetric mode, we map the min/max in the float range to the min/max of the integer range. 
+
+The math equation is like:
+$$
+X_{int8} = round(X_{fp32}/S) + Z
+$$
+
+where $X_{fp32}$ is the input matrix, $S$ is the scale factor,  $Z$ is the integer zero point.
+
+We choose the minimum value of the matrix elements for the lower clipping point, and the maximum value for the upper clipping point. With the lower and upper bounds, the scale factor is computed as:
+$$
+S = \frac{X_{max} - X_{min}}{2^n-1}
+$$
+Where n is the number of bits used for quantization.
+
+The zero point is computed as:
+$$
+Z = - round(X_{min}/S)
+$$
+**Symmetric Quantization**
+
+In symmetric mode, instead of mapping the exact min/max of the float range to the quantized range, we choose the maximum absolute value between min/max.  Therefore, we don`t use a zero-point. The formula is:
+$$
+X_{int8} = round(X_{fp32}/S) \\
+S = \frac{max(|x_{max}|, |x_{min}|)}{2^n -1}
+$$
+
+## SmoothQuant
+
+Some models, especially LLMs, activations are much harder to quantize due to the outliers than weights which the distribution is uniform and flat. The variance amongst the channels for a given token is large but small between magnitudes of a given channels, therefore, the quantization error will decrease if we can use activation per-channel quantization. However, can't perform channel-wise activation quantization currently because it cannot map to hardware-accelerate GEMM kernels well.
+
+SmoothQuant is an alternative method of per-channel activation quantization. It divide the input activation by a per-channel smoothing factor $s\in\mathbb R^{C_i} $ , where $C_i$ is the input channel. Also scale the weights accordingly to keep the mathematical equivalence.
+$$
+Y = (Xdiag(s)^{-1})\cdot(diag(s)W) = \hat{X}\hat{W}
+$$
+This formula migrate the difficulty from activations to weights. In order to control how much difficulty shift to weights, we use a hyper-parameter named migration strength $\alpha$. 
+$$
+s_j = max(|X_j|)^\alpha / max(|W_j|)^{1-\alpha}
+$$
+$j = 1, 2, ...s, C_i$ where j correspond to j-th input channel.
+
+For most of models, such as OPT and BLOOM, $\alpha = 0.5$ which means balance the difficult between activations and weights, can have a low quantization error. For others models with more significant outliers in activations, increase $\alpha$ to a bigger num, for example 0.75 ,  to migrate more quantization difficulty to weights.
+
+## SmoothQuant Support Matrix
+
+| **Algorithm** | **PyTorch** | **TensorFlow** |
+| ------------- | ----------- | -------------- |
+| SmoothQuant   | ✔           | ✖              |
```

Comment:
Please fix the flag
![image](https://user-images.githubusercontent.com/51692656/225833950-e08a0b89-1b1e-477d-8914-bdbfd58d8f03.png)